src/Persistence_matrix/include/gudhi/Persistence_matrix/columns/unordered_set_column.h

```diff
@@ -0,0 +1,1008 @@
+template <class Cell_range>
+inline bool Unordered_set_column<Master_matrix, Cell_constructor>::_multiply_and_add(const Field_element_type& val,
+                                                                                     const Cell_range& column) 
+{
+  if (val == 0u) {
+    if constexpr (Master_matrix::isNonBasic && !Master_matrix::Option_list::is_of_boundary_type) {
+      throw std::invalid_argument("A chain column should not be multiplied by 0.");
+      // this would not only mess up the base, but also the pivots stored.
+    } else {
+      clear();
+      for (const Cell& v : column) {
+        _insert_cell(v.get_element(), v.get_row_index(), column_.end());
+      }
+      return true;
+    }
+  }
+
+  // because the column is unordered, I don't see a way to do both operations in one go...
+  operator*=(val);
+  return _add(column);
+}
+
+template <class Master_matrix, class Cell_constructor>
+template <class Cell_range>
+inline bool Unordered_set_column<Master_matrix, Cell_constructor>::_multiply_and_add(const Cell_range& column,
+                                                                                     const Field_element_type& val) 
+{
+  if (val == 0u) {
+    return false;
+  }
+
+  bool pivotIsZeroed = false;
+
+  for (const Cell& cell : column) {
+    auto it1 = column_.find(const_cast<Cell*>(&cell));
+    if (it1 != column_.end()) {
+      (*it1)->get_element() = operators_->multiply_and_add(cell.get_element(), val, (*it1)->get_element());
+      if ((*it1)->get_element() == Field_operators::get_additive_identity()) {
+        if constexpr (Master_matrix::isNonBasic && !Master_matrix::Option_list::is_of_boundary_type) {
+          if ((*it1)->get_row_index() == chain_opt::get_pivot()) pivotIsZeroed = true;
+        }
+        _delete_cell(it1);
+      } else {
+        if constexpr (Master_matrix::Option_list::has_row_access) ra_opt::update_cell(**it1);
+      }
+    } else {
+      _insert_cell(operators_->multiply(cell.get_element(), val), cell.get_row_index(), column_.end());
+    }
+  }
+
+  return pivotIsZeroed;
+}
+
+}  // namespace persistence_matrix
+}  // namespace Gudhi
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Hash method for @ref Gudhi::persistence_matrix::Unordered_set_column.
+ * 
+ * @tparam Master_matrix Template parameter of @ref Gudhi::persistence_matrix::Unordered_set_column.
+ * @tparam Cell_constructor Template parameter of @ref Gudhi::persistence_matrix::Unordered_set_column.
+ */
+template <class Master_matrix, class Cell_constructor>
+struct std::hash<Gudhi::persistence_matrix::Unordered_set_column<Master_matrix, Cell_constructor> > 
+{
+  size_t operator()(
+      const Gudhi::persistence_matrix::Unordered_set_column<Master_matrix, Cell_constructor>& column) const {
+    std::size_t seed = 0;
+    for (const auto& cell : column) {
+      seed ^= std::hash<unsigned int>()(cell.get_row_index() * static_cast<unsigned int>(cell.get_element())) +
```

Comment:
It does'nt. I naively assumed that the order would be the same if the contained values are the same, but now that I think about it, depending on the number of buckets and the order the element are inserted, the order could be different from one column to another even if the content is the same. Do you know a good hash method for unordered sets?

Reply:
Maybe just the xor (or sum) of the hash of the elements? For containers like vector this is considered bad because it gives the same hash if you permute the elements. But here, permutation invariance looks like a good thing.

Reply:
Wasn't there a lot of collision with xor and sum? But I guess, it is good enough for now.

Reply:
If the hash used for the elements is "good" (which may not be the case of std::hash), I don't see why the xor/sum should have more collisions than something else.
The alternative would be to sort the elements and then hash as usual, but that looks costly.